examples/core/Graphormer/main.py

```diff
@@ -0,0 +1,272 @@
+"""
+This script finetunes and tests a Graphormer model (pretrained on PCQM4Mv2)
+for graph classification on ogbg-molhiv dataset.
+
+Paper: [Do Transformers Really Perform Bad for Graph Representation?]
+(https://arxiv.org/abs/2106.05234)
+
+This flowchart describes the main functional sequence of the provided example.
+main
+│
+└───> train_val_pipeline
+      │
+      ├───> Load and preprocess dataset
+      │
+      ├───> Download pretrained model
+      │
+      ├───> train_epoch
+      │     │
+      │     └───> Graphormer.forward
+      │
+      └───> evaluate_network
+            │
+            └───> Graphormer.inference
+"""
+import argparse
+import random
+
+import torch as th
+import torch.nn as nn
+from accelerate import Accelerator
+from dataset import MolHIVDataset
+from dgl.data import download
+from dgl.dataloading import GraphDataLoader
+from model import Graphormer
+from ogb.graphproppred import Evaluator
+from transformers.optimization import (
+    AdamW,
+    get_polynomial_decay_schedule_with_warmup,
+)
+
+
+def train_epoch(model, optimizer, data_loader, lr_scheduler):
+    model.train()
+    epoch_loss = 0
+    list_scores = []
+    list_labels = []
+    loss_fn = nn.BCEWithLogitsLoss()
+    for iter, (
+        batch_labels,
+        attn_mask,
+        node_feat,
+        in_degree,
+        out_degree,
+        path_data,
+        dist,
+    ) in enumerate(data_loader):
+        optimizer.zero_grad()
+        device = accelerator.device
+
+        batch_scores = model(
+            node_feat.to(device),
+            in_degree.to(device),
+            out_degree.to(device),
+            path_data.to(device),
+            dist.to(device),
+            attn_mask=attn_mask,
+        )
+
+        loss = loss_fn(batch_scores, batch_labels.float())
+
+        accelerator.backward(loss)
+        optimizer.step()
+        lr_scheduler.step()
+        epoch_loss += loss.item()
+        list_scores.append(batch_scores)
+        list_labels.append(batch_labels)
+
+        # release GPU memory
+        del (
+            batch_labels,
+            batch_scores,
+            loss,
+            attn_mask,
+            node_feat,
+            in_degree,
+            out_degree,
+            path_data,
+            dist,
+        )
+        th.cuda.empty_cache()
+
+    epoch_loss /= iter + 1
+
+    evaluator = Evaluator(name="ogbg-molhiv")
+    epoch_train_metric = evaluator.eval(
```

Comment:
In train method, just epoch_metric is enough.
```suggestion
    epoch_metric = evaluator.eval(
```